config/oidc_backends_config.xml.sample

```diff
@@ -1,9 +1,61 @@
 <?xml version="1.0"?>
+<!--
+
+This file shall be properly set for user authentication and authorization leveraging OpenID Connect.
+Here we provide a quick reference for configuration setters, and we provide a complete documentation
+at the following page:
+
+    https://galaxyproject.org/admin/authentication/config/
+
+
+
+Quick Reference
+_______________
+
+- provider:         Sets the name of OpenID Connect (OIDC) Identity Provider (IdP).
+                    The following IdPs are currently supported:
+
+                        { "Google", "Globus" }
+
+- client id:        Sets the id of this client (i.e., your Galaxy instance) with the IdP, which is
+                    obtained from the IdP at client registration.
+                    See: https://tools.ietf.org/html/rfc6749#section-2.2
+
+- client secret:    A secret generated by IdP for your client upon its registration.
+                    See https://tools.ietf.org/html/rfc6749#section-2.3.1
+
+
+
+IMPORTANT NOTES
+_______________
+
+While registering your client (i.e., your Galaxy instance), you would need to provide the IdP with a
+`redirect_uri`: a URL at which the IdP will callback your Galaxy instance upon successful user authentication.
+
+This URL for your Galaxy instance is composed as:
+
+    <host_url>/authnz/<provider>/callback
+
+
+Some examples:
+
+- using localhost to authenticate with Google:
+    http://localhost:8080/authnz/google/callback
+
+- using localhost to authenticat with Globus:
+    http://localhost:8080/authnz/globus/callback
+
+- using an instance hosted at `https://usegalaxy.org` with Google:
+    https://usegalaxy.org/authnz/google/callback
+
+
+Please mind `http` and `https`.
+
+-->
 <OIDC>
     <provider name="Google">
         <client_id> ... </client_id>
         <client_secret> ... </client_secret>
-        <redirect_uri>http://localhost:8080/authnz/google/callback</redirect_uri>
```

Comment:
Why was this dropped? This tag stayed for the globus config.

Reply:
I guess I missed it, I will remove it for Globus as well. We're dropping it mainly because admins do not have any "freedom" setting up this setter (e.g., they cannot change `/authnz/google/callback` to `/authentication/another_callback/google`. We had it user defined, because (a) they must know this and set it correctly when they're registering their Galaxy with the idp, and (b) it was not clear what is the best method for getting the base URL (i.e., the `http://localhost:8080/` part of the sample). 

However, in this PR, we're updating docs making sure admins know how to set this url with the idp (reg. (a)); and we think `trans.request.host_url` is our best option for getting the base URL (reg. (b)). 

Hence, I think we can drop this setter, and make config easier.